prov/verbs/src/fi_verbs.h

```diff
@@ -484,6 +484,17 @@ int vrb_srq_context(struct fid_domain *domain, struct fi_rx_attr *attr,
 int vrb_domain_xrc_init(struct vrb_domain *domain);
 int vrb_domain_xrc_cleanup(struct vrb_domain *domain);
 
+enum ibv_gid_type {
+	IBV_GID_TYPE_IB_ROCE_V1,
+	IBV_GID_TYPE_ROCE_V2,
+};
+
+#if !HAVE_DECL_IBV_QUERY_GID_TYPE && HAVE_IBV_QUERY_GID_TYPE
+	extern int *ibv_query_gid_type(struct ibv_context *context,
+					uint8_t port_num, unsigned int index,
+					enum ibv_gid_type *gid_type);
```

Comment:
We need some sort of implementation to avoid an unresolved symbol error.  Please test the build with an older version of verbs where this function does not exist.

Reply:
The function is being declared (extern shouldn't be needed), but it's never defined.